### PR TITLE
Scripts/Spells: activate warrior spell Victory Rush

### LINKED ISSUE #19281: Core/Spell Victory Rush activation problems.
**Description:**  

The spell Victory Rush (ID:34428) presents a problem in activation.  

**Current behaviour: (Tell us what happens.)**  

The spell came active in any spec.  

**Expected behaviour: (Tell us what should happen instead.)**  

This Spell should only activate in spec Protection.  

**Steps to reproduces the problem:**  

1- Create a Warrior and levelup 15  
2- Kill an enemy that give you xp in any spec.  

**Branch(es):** Master  

**TC rev. hash/commit:**  

a36f34af  

**TDB version:** 7.0.3+updates  

**Operating system:** windows 10

### PR DESCRIPTION
**Pull request by HannibalRoG**
[//]: # (***************************)
[//]: # (** FILL IN THIS TEMPLATE **)
[//]: # (***************************)

**Changes proposed:**

-  Fix for activation of Victory Rush. It only activate on spec protection and arms, not in fury.
-  Implement Impending Victory (Talent level 30 spec Protection)
Thanks to @DoctorKraft 

**Target branch(es):** 3.3.5/master

- [x] master

**Issues addressed:** Closes #  (insert issue tracker number)

Closes #19281

**Tests performed:** (Does it build, tested in-game, etc.)
Compiled and tested in game.

### CODE REVIEW

**Comment by DoctorKraft** [comment]
Alphabetical order is bad for spellscript but it's ok for spellregister :)
Also missing new line in SQL File ^^

**Comment by tkrokli** [comment]
Added missing `}, }, };` at the end of `class spell_warr_impending_victory` section + merged 1 spell enum from master into the .cpp file.

**Comment by DoctorKraft** [comment]
`spell_warr_victorious_state` not present in sql ^^

**Comment by tkrokli** [comment]
Thanks. Will fix.

**Comment by DoctorKraft** [comment]
Oops, I did not see but with merge register and return are missing for `spell_warr_victorious_state`
https://github.com/TrinityCore/TrinityCore/pull/19297/commits/e650f9abe70a37b4cf1d6d19843e76d66952087c#diff-365c72c0309d2d13106c5c6de4a88ce7R983

**Comment by HannibalRoG** [comment]
What about this PR, any news?

**Comment by tkrokli** [comment]
@HannibalRoG : according to @DoctorKraft 
> `register` and `return` are missing for `spell_warr_victorious_state`

**Comment by Aokromes** [comment]
/home/travis/build/TrinityCore/TrinityCore/src/server/scripts/Spells/spell_warrior.cpp:443:17: fatal error: 
      use of undeclared identifier 'SPELL_WARRIOR_IMPENDING_VICTORY_HEAL'
                SPELL_WARRIOR_IMPENDING_VICTORY_HEAL
                ^

File: src/server/scripts/Spells/spell_warrior.cpp [hunk 1]
```diff
@@ -1128,77 +1166,46 @@ class spell_warr_victory_rush : public SpellScriptLoader
...
+        {
+            return ValidateSpellInfo
+            ({
+                SPELL_WARRIOR_IMPENDING_VICTORY
+            });
+        }
+        
+        void HandleOnProc(AuraEffect const* /*aurEff*/, ProcEventInfo& /*procInfo*/)
+        {
+            if (Unit* caster = GetCaster())
```
Comment (Seyden) [inline src/server/scripts/Spells/spell_warrior.cpp]: Can't be null for this aura

**Comment by tkrokli** [comment]
`SPELL_WARRIOR_IMPENDING_VICTORY_HEAL` does not have any enum value.
You need to add it to the enum list with a spell ID to make the script compile.

**Comment by Aokromes** [comment]
Conflicting files
src/server/scripts/Spells/spell_warrior.cpp

**Comment by tkrokli** [comment]
- removed the merge conflict based on outdated spell / code overlap in the source file
- added missing enumeration

File: src/server/scripts/Spells/spell_warrior.cpp [hunk 2]
```diff
@@ -1166,77 +1207,46 @@ class spell_warr_victory_rush : public SpellScriptLoader
...
         }
+    };
+
+    AuraScript* GetAuraScript() const override
+    {
+        return new spell_warr_victorious_state_Aurascript();
+    }
 };
 
 // 50725 Vigilance
```
Comment (tkrokli) [inline src/server/scripts/Spells/spell_warrior.cpp]: Replaced by spell 122710 ?
- http://www.wowhead.com/spell=122710/vigilance

**Pull request closed without merging**